system-test/tpc-test.ts

```diff
@@ -0,0 +1,237 @@
+
+    const query = {
+      columns: ['VenueId', 'VenueName', 'Capacity'],
+      keySet: {
+        all: true,
+      },
+    };
+
+    console.log(`Reading rows in the table ${table.name}`);
+
+    const [rows] = await venuesTable.read(query);
+
+    rows.forEach(row => {
+      const json = row.toJSON();
+      console.log(
+        `VenueId: ${json.VenueId}, VenueName: ${json.VenueName}, Capacity: ${json.Capacity}`,
+      );
+    });
+
+    console.log(`deleting table ${table.name}`);
+    await table.delete();
+    console.log(`deleted table ${table.name}`);
+  }
+
+  after(async () => {
+    // TODO: Solve the issue where tests fail because instances don't get created on time.
+    // Notes: Creating instances can take time and if they are not ready in
+    // time then tests can fail. This shouldn't happen because if the create
+    // instance long running operation completes then the instance should be
+    // ready and shouldn't produce the `Error: 5 NOT_FOUND` error.
+    // Uncomment the code below when the task above is addressed:
+    // const instance = bigtable.instance(instanceId);
+    // await instance.delete({});
+  });
+
+  describe('set universe with spanner client option', () => {
+    it('should set universeDomain option', async () => {
+      const universeDomain = UNIVERSE_DOMAIN_CONSTANT;
+      const options = {
+        projectId,
+        universeDomain,
+      };
+      const spanner = new Spanner(options);
+      const instanceId = 'test-instance';
+      const databaseId = 'test-database';
+
+      try {
+        await runTest(spanner, instanceId, databaseId);
+      } catch (e) {
+        assert.ifError(e);
+      }
+    });
+
+    it('should set universe_domain option', async () => {
+      const universe_domain = UNIVERSE_DOMAIN_CONSTANT;
+      const options = {
+        projectId,
+        universe_domain,
+      };
+      const spanner = new Spanner(options);
+      const instanceId = 'test-instance';
+      const databaseId = 'test-database';
+
+      try {
+        await runTest(spanner, instanceId, databaseId);
+      } catch (e) {
+        assert.ifError(e);
+      }
+    });
+
+    it('set both universe_domain and universeDomain option with the same domain value', async () => {
+      const universeDomain = UNIVERSE_DOMAIN_CONSTANT;
+      const universe_domain = UNIVERSE_DOMAIN_CONSTANT;
+      const options = {
+        projectId,
+        universeDomain,
+        universe_domain,
+      };
+      const spanner = new Spanner(options);
+      const instanceId = 'test-instance';
+      const databaseId = 'test-database';
+
+      try {
+        await runTest(spanner, instanceId, databaseId);
+      } catch (err) {
+        assert.ifError(err);
+      }
+    });
+
+    it('set both universe_domain and universeDomain option with the different domain value', async () => {
+      const fakeError = new Error(
+        'Please set either universe_domain or universeDomain, but not both.',
+      );
```

Comment:
Is this error only thrown when both are set with different value, if value are same then it works ?

Reply:
yes, the error will only get thrown when both the properties `universe_domain` and `universeDomain` are set but with different values
ref: https://github.com/googleapis/nodejs-spanner/blob/main/src/v1/spanner_client.ts#L119-#L127